### PR TITLE
Move retry, enforce_for_root, local_users_only options to pwquality.conf

### PR DESCRIPTION
**Pull request by redhatrises**
- retry, enforce_for_root, local_users_only options can now be configured in pwquality.conf
- Fixes RHBZ#1537240

### CODE REVIEW

File: src/pwquality.conf [hunk 1]
```diff
@@ -61,3 +61,15 @@
 #
 # Path to the cracklib dictionaries. Default is to use the cracklib default.
 # dictpath =
+#
+# Prompt user at most N times before returning with error. The default is 1.
+retry = 3
```
Comment (t8m) [inline src/pwquality.conf]: Please do not set non-default value here.

File: src/settings.c [hunk 2]
```diff
@@ -68,7 +71,10 @@ static const struct setting_mapping s_map[] = {
  { "usercheck", PWQ_SETTING_USER_CHECK, PWQ_TYPE_INT},
  { "enforcing", PWQ_SETTING_ENFORCING, PWQ_TYPE_INT},
  { "badwords", PWQ_SETTING_BAD_WORDS, PWQ_TYPE_STR},
- { "dictpath", PWQ_SETTING_DICT_PATH, PWQ_TYPE_STR}
+ { "dictpath", PWQ_SETTING_DICT_PATH, PWQ_TYPE_STR},
+ { "retry", PWQ_SETTING_RETRY_TIMES, PWQ_TYPE_INT},
+ { "enforce_for_root", PWQ_SETTING_ENFORCE_ROOT, PWQ_TYPE_INT},
```
Comment (t8m) [inline src/settings.c]: This should be PWQ_TYPE_SET.

File: src/settings.c [hunk 3]
```diff
@@ -68,7 +71,10 @@ static const struct setting_mapping s_map[] = {
...
+ { "local_users_only", PWQ_SETTING_LOCAL_USERS, PWQ_TYPE_INT}
```
Comment (t8m) [inline src/settings.c]: And this one too.

File: src/settings.c [hunk 4]
```diff
@@ -36,6 +36,9 @@ pwquality_default_settings(void)
         pwq->dict_check = PWQ_DEFAULT_DICT_CHECK;
         pwq->user_check = PWQ_DEFAULT_USER_CHECK;
         pwq->enforcing = PWQ_DEFAULT_ENFORCING;
+	pwq->retry_times = PWQ_DEFAULT_RETRY_TIMES;
+	pwq->enforce_for_root = PWQ_DEFAULT_ENFORCE_ROOT;
+	pwq->local_users_only = PWQ_DEFAULT_LOCAL_USERS;
```
Comment (t8m) [inline src/settings.c]: Please align the indentation here.

File: src/pwquality.conf [hunk 5]
```diff
@@ -61,3 +61,15 @@
...
+#
+# Enforces pwquality checks on the root user password.
+# The check is disabled if the value is 0.
+# enforce_for_root = 1
```
Comment (t8m) [inline src/pwquality.conf]: Correctly commented out, but there should not be any '= 1' And the text should be "Enabled if the option is present.'

File: src/pwquality.conf [hunk 6]
```diff
@@ -61,3 +61,15 @@
...
+#
+# Skip testing the password quality for users that are not present in the
+# /etc/passwd file.
+# The check is disabled if the value is 0.
+# local_users_only = 0
```
Comment (t8m) [inline src/pwquality.conf]: Again no '= 0' here. And  text above should be 'Enabled if the option is present.'

File: src/pwquality.conf (hunk 1 above)
Comment (redhatrises) [reply src/pwquality.conf]: Fixed

File: src/settings.c (hunk 2 above)
Comment (redhatrises) [reply src/settings.c]: Fixed

File: src/settings.c (hunk 3 above)
Comment (redhatrises) [reply src/settings.c]: Fixed

File: src/settings.c (hunk 4 above)
Comment (redhatrises) [reply src/settings.c]: I guess I missed a few as I thought I had made sure the indentation was right. Oh well. Fixed.

File: src/pwquality.conf (hunk 5 above)
Comment (redhatrises) [reply src/pwquality.conf]: Fixed

File: src/pwquality.conf (hunk 6 above)
Comment (redhatrises) [reply src/pwquality.conf]: Fixed

**Comment by redhatrises** [comment]
@t8m updated PR with requested changes.

**Comment by t8m** [comment]
Thanks for the PR.

**Comment by redhatrises** [comment]
Thanks for review @t8m!

**Pull request merged**